### PR TITLE
chore: switch to pnpm

### PR DESCRIPTION
**Pull request by Shinigami92**

### CODE REVIEW

**Comment by Shinigami92** [comment]
Behaviour copied from https://github.com/vitejs/vite/blob/ea615b07ce8c5dc55d24f293803c7f5886ce425b/netlify.toml and https://github.com/vitejs/vite/blob/ea615b07ce8c5dc55d24f293803c7f5886ce425b/package.json

**Comment by Shinigami92** [comment]
Need to install `npm-run-all`

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,22 +12,28 @@ jobs:
     name: 'Testing'
     steps:
       - uses: actions/checkout@v2
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 6
       - uses: actions/setup-node@v2
         with:
           node-version: '16'
```
Comment (prisis) [inline .github/workflows/ci.yml:20]: Should this not us a matrix variable with all supported node version?

Comment (Shinigami92) [reply .github/workflows/ci.yml:20]: Totally, feel free to open a new PR based on main for this
I also can help you with that if you need
You could orient yourself a bit ob Vite's ci.yml

If you won't open a PR for that, I already planned it, but didn't do it right now.

**Comment by damienwebdev** [comment]
Why can't we just use what Node provides out of the box with `npm` do we really need `pnpm`?

**Comment by Shinigami92** [comment]
> Why can't we just use what Node provides out of the box with `npm` do we really need `pnpm`?

pnpm has MANY improvements over npm
It downloads the files in a global local cache + it uses symlinks to that
So your e.g. SSD will love you

npm is downloading everything always when you removed the node_modules folder and try to make a clean install

**Comment by prisis** [comment]
Wouldn't it be easier for new contributor to use yarn? If i'm not mistaken its comes with node

**Comment by Shinigami92** [comment]
> Wouldn't it be easier for new contributor to use yarn? If i'm not mistaken its comes with node

yarn has some benefits but is not well maintained
also it is driven by facebook where pnpm is community driven
also yarn v3 needs some more configuration where pnpm doesn't need anything
new contributors are directly informed and get telled that they need to just do `npm install -g pnpm`
pnpm is also in corepack compatible
yarn v1 is legacy and not maintained anymore but you need it to switch to v3
pnpm has a much better CLI

**Comment by prisis** [comment]
Im pretty fine with pnpm, using it too in most all of my projects

**Review by prisis — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Shinigami92) [reply .github/workflows/ci.yml:20]: Issue #159 created

**Review by import-brain — APPROVED** [review]

**Comment by Shinigami92** [comment]
@prisis I rebase this branch, please re-approve it :slightly_smiling_face:

**Review by import-brain — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Review by JessicaSachs — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**